### PR TITLE
Allow for parsing of sequential ranges

### PR DESCRIPTION
**Pull request by JohannesKlauss**
Currently the library cannot parse a sequential range where the rank of the first card is constant, something like `A5-A2s`. This addition allows to parse that.

If I should change anything, please let me know 🙂

### CODE REVIEW

File: src/holdem/parse.rs [hunk 1]
```diff
@@ -666,11 +692,27 @@ mod test {
...
     }
 
+    #[test]
+    fn test_parse_sequential() {
+        assert_eq!(
+            RangeParser::parse_one(&String::from("A9s-A5s"))
+                .unwrap()
+                .len(),
+            20
+        );
```
Comment (elliottneilclark) [inline src/holdem/parse.rs:702]: Please add a test showing that each hand satisfies the invariants.

- Every hand has an Ace in it.
- The other card is between 9 and 5
- Both cards are the same suit

File: src/holdem/parse.rs [hunk 2]
```diff
@@ -69,6 +69,8 @@ enum RangeIterValueSpecifier {
     Static(Value),
     /// Pair
     Pair,
+    /// Sequential. Value is the higher value. Second value is the gap.
+    Sequential(Value, u8),
```
Comment (elliottneilclark) [inline src/holdem/parse.rs:73]: I  think `StaticWithRange` or `StaticRange` are better names. These aren't hands where the cards are sequential.

**Review by elliottneilclark — CHANGES_REQUESTED** [review]

**Comment by elliottneilclark** [comment]
I really like this feature. Thank you other than naming and a little extra testing this looks good.

**Comment by JohannesKlauss** [comment]
> I really like this feature. Thank you other than naming and a little extra testing this looks good.

I added the requested changes. Can you tell me what code format analyzer you are using? Doesn't state it in the CLI output.

**Review by elliottneilclark — APPROVED** [review]

**Comment by elliottneilclark** [comment]
> Can you tell me what code format analyzer you are using?

For the testing, we run the equivalent of the following:

```
cargo fmt --all
cargo clippy --all-targets --all
cargo test --all --all-targets
cargo audit
```

**Comment by JohannesKlauss** [comment]
@elliottneilclark Would it be possible to publish a new release, so I can use that feature 🙂 ?

**Comment by elliottneilclark** [comment]
@JohannesKlauss `v3.0.0-beta.8` is released.

**Pull request merged**